### DISCUSSION TITLE
How to have words in Quarto blog urls separated by hyphens and not "%20"

### DISCUSSION THREAD

**Discussion by paul-kelleher** [quarto-dev/quarto-cli#8790] [Q&A] [answered]
I'm sure this has been asked and answered, but I can't seem to find it. I recently created a Quarto blog that is hosted on Github pages. Currently the urls of blog posts have title words separated by "%20". What setting(s) do I change to have title words in urls separated by hyphens instead?

Thanks!

**Comment by mcanouil** [comment, marked as answer]
Do you use Quarto documents using spaces in their file names?
The URLs uses the name of the files.

**Reply by paul-kelleher** [reply]
I think I just figured it out (or at least *a* solution): Each blog post has its own folder. The url uses the name of the folder, which can be different from the title of the blog post. So I renamed the folder "hyphenated-url-test" but left the actual yaml of the post untouched (e.g. the title field is `title: "Hyphenated Url Test"`). And that did the trick.

**Reply by mcanouil** [reply]
The url is based on file name and directories.
I am pretty sure it is never built from the title in the YAML (I might be wrong here).

Note that it's always a good thing to not have spaces in directory and file names.
